src/crafting.rs

```diff
@@ -174,6 +180,17 @@ pub struct CraftingRecipe {
     pub(crate) produces: ItemType,
 }
 
+impl Inventory {
+    pub fn ingredients_available(&mut self, recipe: &CraftingRecipe) -> bool {
+        for ingredient in recipe.needed.clone() {
```

Comment:
Clone recipe or clone every ingredient ? Which is the best way to manage memory and possession ? Beginner's curiosity

Reply:
I actually don't think we even need a clone here 
```rust
pub fn ingredients_available(&mut self, recipe: &CraftingRecipe) -> bool {
    for ingredient in &recipe.needed {
        if !self.can_remove(ingredient) {
            return false;
        }
    }
    true
}